Lib/test/test_cmd_line.py

```diff
@@ -337,6 +337,13 @@ def test_osx_android_utf8(self):
         self.assertEqual(p.returncode, 0)
 
     def test_non_interactive_output_buffering(self):
+        PYTHONUNBUFFERED = "PYTHONUNBUFFERED"
+        python_unbuffered_value = os.environ.get(PYTHONUNBUFFERED, "0")
+        python_unbuffered_on = python_unbuffered_value != "0"
+        if python_unbuffered_on:
+            # we expect buffered stdio
+            os.environ[PYTHONUNBUFFERED] = "0"
```

Comment:
Out of curiosity, but can't you use some `mock.patch.dict` or something like this? because, we're anyway disabling PYTHONUNBUFFERED. So we can just always put it to 0 in this test, right?

Reply:
I will investigate